server/src/modules/maps/maps.service.ts

```diff
@@ -248,6 +248,39 @@ export class MapsService {
         return mapDB.id;
     }
 
+    async update(mapID: number, userID: number, updateBody: UpdateMapDto): Promise<void> {
+        const map = await this.mapRepo.get(mapID);
```

Comment:
I don't know if there are status flags that not every user is allowed to set. If there are then a check needs to be added here. 
As far as I could see the old API doesn't check for it so I didn't either

Reply:
Yeah the permissions system here does seem a bit odd, let's just implement it the same way for now.

Though one thing you're missing, the old API is changing that the map is not in the `REJECTED` or `REMOVED` state, let's at least add that. So just return a 400 if it's in either of those states.